src/graph/GraphViewServer.ts

```diff
@@ -153,6 +162,24 @@ export class GraphViewServer extends EventEmitter {
     });
   }
 
+  private handleGetPageState() {
+    console.log('getPageState');
+
+    if (this._lastQuery) {
+      this._socket.emit('setPageState', this._lastQuery, this._lastErrorMsg, this._lastResults, this._lastView);
+    }
+  }
+
+  private handleSetQuery(query: string) {
+    console.log('setQuery');
+    this._lastQuery = query;
```

Comment:
This means that if you query for results, wait for the results to show, then type a new query (or the start of a new query), but don't hit Execute, then tab away and back, the new query will show, and the old results will animate in, making it look like they're the new results.  Correct?

Perhaps you should throw away the last results (in the server, not the client) when the query changes?